### PR TITLE
Set os_version to match the output of oslevel -s

### LINKED ISSUE #693: Incorrect output of os_version on AIX
### Version

    # lslpp -L chef
      Fileset                      Level  State  Type  Description (Uninstaller)
      ----------------------------------------------------------------------------
      chef                      12.5.1.1    C     F    The full stack of chef
    # ohai --version
    Ohai: 8.7.0

### Environment

AIX 6.1 or 7.1

### Steps to reproduce

    # ohai os
    [
      "aix"
    ]
    # ohai os_version
    [
      "1"
    ]

### Expected result

    # ohai os_version
    [
      "6.1"
    ]

or

    # ohai os_version
    [
      "6.1.0.0"
    ]

or

    # ohai os_version
    [
      "6100-07"
    ]

or 

    # ohai os_version
    [
      "6100-07-00-0000"
    ]

### Actual result

is always 1. 

The file /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/ohai-8.7.0/lib/ohai/plugins/aix/kernel.rb uses uname command to determine the version and the release of AIX. The AIX command uname -v, which supposes to return the version, returns in fact release - 1 from 6.1 or 7.1. The proper command to determine release level of AIX is oslevel. Without any options it returns the value I personally would expect as os_version in ohio, e.g.:

    # oslevel
    6.1.0.0

with the option *-r* it returns "release", in fact - technology level of AIX. E.g. for AIX 6.1 TL7 it is:

    # oslevel -r
    6100-07

with the option *-s* it returns the latest installed service pack. E.g. for AIX 6.1 TL7 SP0 it is:

    # oslevel -s
    6100-07-00-0000

### PR DESCRIPTION
**Pull request by juliandunn**
Fixes #693.

### CODE REVIEW

File: lib/ohai/plugins/aix/os.rb [hunk 1]
```diff
@@ -25,6 +25,6 @@
 
   collect_data(:aix) do
     os collect_os
-    os_version kernel[:version]
+    os_version shell_out("oslevel -s").stdout.split($/)[0]
```
Comment (tas50) [inline lib/ohai/plugins/aix/os.rb]: What's the split accomplish?

Comment (juliandunn) [inline lib/ohai/plugins/aix/os.rb]: It's to hedge against whitespace appearing after the command output. Sometimes IBM utilities are not clean on this front.

Comment (mcquin) [inline lib/ohai/plugins/aix/os.rb]: Could you add or modify the the existing test to include a shell out result with trailing whitespace?

Comment (tas50) [inline lib/ohai/plugins/aix/os.rb]: ^

Comment (juliandunn) [inline lib/ohai/plugins/aix/os.rb]: Done. By whitespace I meant linebreak :-)

Comment (tas50) [inline lib/ohai/plugins/aix/os.rb]: So any reason not to just use .strip here?

Comment (juliandunn) [inline lib/ohai/plugins/aix/os.rb]: Could do that too; I was just copying the style from the rest of the AIX plugins. You'd prefer `.strip` then I take it?

Comment (tas50) [inline lib/ohai/plugins/aix/os.rb]: It's a lot more clear what you're trying to do with .strip

**Comment by thommay** [comment]
:+1:

**Comment by tas50** [comment]
:+1:
